File: backend/todo/src/main/java/com/example/todo/models/TodoList.java
package com.example.todo.models;

import jakarta.persistence.*;
import java.util.List;

@Entity
@Table(name = "todo_lists")
public class TodoList {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    
    @Column(nullable = false)
    private String title;
    
    @Column(name = "is_shared", nullable = false)
    private Boolean isShared = false;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;
    
    @OneToMany(mappedBy = "list", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<TodoItem> items;
    
    @OneToMany(mappedBy = "list", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Collaboration> collaborations;
    
    // Constructors
    public TodoList() {}
    
    // Getters and Setters
    public Integer getId() { return id; }
    public void setId(Integer id) { this.id = id; }
    
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    
    public Boolean getIsShared() { return isShared; }
    public void setIsShared(Boolean isShared) { this.isShared = isShared; }
    
    public User getOwner() { return owner; }
    public void setOwner(User owner) { this.owner = owner; }
    
    public List<TodoItem> getItems() { return items; }
    public void setItems(List<TodoItem> items) { this.items = items; }
    
    public List<Collaboration> getCollaborations() { return collaborations; }
    public void setCollaborations(List<Collaboration> collaborations) { this.collaborations = collaborations; }
}

File: frontend/todo/src/structures/todoList/todoList.ts
import type User from "../user/user.ts";
import type todoItem from "../todoItem/todoItem.ts";

export default class todoList{
    id:number;
    title:string;
    isShared:boolean;
    owner:User;
    todoItemList?:todoItem[];

    constructor(
        id:number,
        title:string,
        isShared:boolean,
        owner:User
    ) {
        this.id = id;
        this.title = title;
        this.isShared = isShared;
        this.owner = owner;
    }

    setTodoItemList(todo_item_list:todoItem[]):void{
        this.todoItemList = todo_item_list;
    }

    getTodoItemList():todoItem[]{
        return this.todoItemList || [];
    }

}